src/ui/common/src/components/MultiFileViewer/index.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  faChevronDown,
+  faChevronRight,
+} from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Editor from '@monaco-editor/react';
+import { TreeItem, TreeView } from '@mui/lab';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import React, { useEffect, useState } from 'react';
+
+import { MenuSidebarOffset } from '../layouts/default';
+
+type Props = {
+  files: Record<string, any>;
+  codeHeight?: string;
+};
+
+const MultiFileViewer: React.FC<Props> = ({ files, codeHeight = '30vh' }) => {
+  const [selectedFile, setSelectedFile] = useState('');
+  const [matches, setMatches] = useState(false);
+
+  useEffect(() => {
+    const media = window.matchMedia('(min-width: 1000px)');
+    if (media.matches !== matches) {
+      setMatches(media.matches);
+    }
+    const listener = () => setMatches(media.matches);
+    window.addEventListener('resize', listener);
```

Comment:
What's going on here with the media query and event listener?
Was this code taken from an example on the mui.com website?